src/common/rdb_stream.cc

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "rdb_stream.h"
+
+#include "fmt/format.h"
+#include "vendor/crc64.h"
+#include "vendor/endianconv.h"
+
+StatusOr<size_t> RdbStringStream::Read(char *buf, size_t n) {
+  if (pos_ + n > input_.size()) {
+    return {Status::NotOK, "unexpected EOF"};
+  }
+  memcpy(buf, input_.data() + pos_, n);
+  pos_ += n;
+  return n;
+}
+
+StatusOr<uint64_t> RdbStringStream::GetCheckSum() const {
+  if (input_.size() < 8) {
+    return {Status::NotOK, "invalid payload length"};
+  }
+  uint64_t crc = crc64(0, reinterpret_cast<const unsigned char *>(input_.data()), input_.size() - 8);
+  memrev64ifbe(&crc);
```

Comment:
It's a bit tricky here. Should we `memrev64ifbe` for the crc64 of the Stream?

Reply:
yes, we should, , that's what Redis does.